pilot/helpers/agents/Developer.py

```diff
@@ -112,6 +112,7 @@ def implement_task(self, i, development_task=None):
                 result['completed_steps'] = task_steps[:step_index]
                 result['current_step'] = task_steps[step_index]
                 result['next_steps'] = task_steps[step_index + 1:]
+                result['current_step_index'] = step_index
 
                 convo_dev_task.remove_last_x_messages(1)
```

Comment:
in Debugger.py you are removing last 2 messages before calling 'update_task.prompt' and here only last one. Not sure if that is ok, please just double check.

Reply:
You're right - good catch

Reply:
Btw, in general, we should almost always remove the last 2 messages when we want to remove the last call from the conversation.